contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"time"
+
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/spf13/viper"
+)
+
+type FlowInfoStatus int32
+
+const (
+	FlowInfoNoState     FlowInfoStatus = 0
+	FlowInfoOneEntry    FlowInfoStatus = 1    // startup state
+	FlowInfoTwoEntries  FlowInfoStatus = 2    // startup state
+	FlowInfoOldFlow     FlowInfoStatus = 3    // standard state
+	FlowInfoOneEntryDup FlowInfoStatus = 4    // error state
+	FlowInfoThreeFlows  FlowInfoStatus = 5    // error state
+	FlowInfoTimedOut    FlowInfoStatus = 6    // error state
+)
+
+// time for cleanup of old connections
+const timeoutPeriodCleanup = 20
+
+// timeout to declare rejected flow
+const timeoutPeriodReject = 3
+
+type singleFlowInfo struct {
+	UUID                     string
+	Flow                     *flow.Flow         // most recent Flow seen for this UUID
+	Saw                      bool               // true if this UUID was seen in most recent interation
+}
+
+type combinedFlowInfo struct {
+	Status                   FlowInfoStatus
+	PrevStatus               FlowInfoStatus
+	MetricReported           *flow.FlowMetric
+	LastUpdateMetricReported *flow.FlowMetric // FIXME - we might not need this
+	Flow1                    singleFlowInfo
+	Flow2                    singleFlowInfo
+}
+
+type accRejAction struct {
+	// table of saved flows
+	accRejState map[string]*combinedFlowInfo
+	timeOfLastCleanup	int64
+}
+
+// NewAction - creates data structures needed for vpc Accept/Reject based on measurements flows at 2 points
+func NewAction(cfg *viper.Viper) (interface{}, error) {
+	accRejState := make(map[string]*combinedFlowInfo)
+	now := time.Now()
+	secs := now.Unix()
+	accRej := &accRejAction{
+		accRejState:  accRejState,
+		timeOfLastCleanup: secs,
+	}
+	return accRej, nil
+}
+
+// initAction - initialize internal data structure for this round
+func (t *accRejAction) initAction() {
```

Comment:
call the method init() or setup() if it were a function the initAction() would make sense

Reply:
OK